hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpAuthenticationMessageHandlerIntegrationTest.java

```diff
@@ -0,0 +1,446 @@
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.context.annotation.Description;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.StringUtils;
+
+import ru.yandex.qatools.allure.annotations.Features;
+import ru.yandex.qatools.allure.annotations.Stories;
+
+/**
+ *
+ */
+@Features("Component Tests - Device Management Federation API")
+@Stories("Amqp Authentication Message Handler")
+public class AmqpAuthenticationMessageHandlerIntegrationTest extends AbstractAmqpIntegrationTest {
+
+    private static final String TARGET_SECRUITY_TOKEN = "12345";
+    private static final String TARGET_TOKEN_HEADER = "TargetToken " + TARGET_SECRUITY_TOKEN;
+    private static final String TENANT_EXIST = "DEFAULT";
+    private static final String TARGET = "NewDmfTarget";
+
+    @Before
+    public void testSetup() {
+        enableTargetTokenAuthentification();
+    }
+
+    @Test
+    @Description("Tests wrong content type. This message is invalid and should not requeued. Additional the receive message is null")
+    public void wrongContentType() {
+        final Message createAndSendMessage = getDmfClient()
+                .sendAndReceive(new Message("".getBytes(), new MessageProperties()));
+        assertThat(createAndSendMessage).isNull();
+    }
+
+    @Test
+    @Description("Tests a null message. This message is invalid and should not requeued. Additional the receive message is null")
+    public void securityTokenIsNull() {
+        final Message createAndSendMessage = sendAndReceiveAuthenticationMessage(null);
+        assertThat(createAndSendMessage).isNull();
+    }
+
+    @Test
+    @Description("Tenant in the message is null. This message is invalid and should not requeued. Additional the receive message is null")
+    public void securityTokenTenantIsNull() {
+        final TenantSecurityToken securityToken = createTenantSecurityToken(null, TARGET,
+                FileResource.createFileResourceBySha1(TARGET_SECRUITY_TOKEN));
+        final Message createAndSendMessage = sendAndReceiveAuthenticationMessage(securityToken);
+        assertThat(createAndSendMessage).isNull();
+    }
+
+    @Test
+    @Description("Target in the message is null.This message is invalid and should not requeued. Additional the receive message is null")
+    public void securityTokenFileResourceIsNull() {
+        enableAnonymousAuthentification();
+        final TenantSecurityToken securityToken = createTenantSecurityToken(TENANT_EXIST, TARGET, null);
+        final Message returnMessage = sendAndReceiveAuthenticationMessage(securityToken);
+        verifyResult(returnMessage, HttpStatus.NOT_FOUND, null);
+    }
+
+    @Test
+    @Description("Verify that login fails if the given credential not match.")
+    public void loginFailedBadCredentials() {
+        tenantConfigurationManagement.addOrUpdateConfiguration(TenantConfigurationKey.ANONYMOUS_DOWNLOAD_MODE_ENABLED,
```

Comment:
Extract to private method `disableAnonymousDownload()` like all the other config methods